diff --git a/node_modules/solid-app-router/dist/index.d.ts b/node_modules/solid-app-router/dist/index.d.ts
index a0cd1c3..4f278d9 100644
--- a/node_modules/solid-app-router/dist/index.d.ts
+++ b/node_modules/solid-app-router/dist/index.d.ts
@@ -5,6 +5,10 @@ export type { Params, QueryParams } from "./recognizer";
 export declare type DataFnParams<T> = {
     params: Params<T>;
     query: QueryParams;
+    root: string;
+    location: string;
+    pending: boolean;
+    level: number;
 };
 export declare type DataFn<T = BaseObject> = (props: DataFnParams<T>) => BaseObject;
 export interface LinkProps extends JSX.AnchorHTMLAttributes<HTMLAnchorElement> {
diff --git a/node_modules/solid-app-router/dist/index.jsx b/node_modules/solid-app-router/dist/index.jsx
index 3f426a0..9f8c3f8 100644
--- a/node_modules/solid-app-router/dist/index.jsx
+++ b/node_modules/solid-app-router/dist/index.jsx
@@ -107,14 +107,7 @@ function createRouter(routes, initialURL, root = "") {
             if (levels[i].handler.data) {
                 data[i] = createRoot(dispose => {
                     disposers[i] = dispose;
-                    return levels[i].handler.data({
-                        get params() {
-                            return state.params;
-                        },
-                        get query() {
-                            return state.query;
-                        }
-                    });
+                    return levels[i].handler.data(state);
                 });
             }
             else
