diff --git a/node_modules/solid-app-router/dist/index.jsx b/node_modules/solid-app-router/dist/index.jsx
index 3f426a0..771a23b 100644
--- a/node_modules/solid-app-router/dist/index.jsx
+++ b/node_modules/solid-app-router/dist/index.jsx
@@ -113,6 +113,9 @@ function createRouter(routes, initialURL, root = "") {
                         },
                         get query() {
                             return state.query;
+                        },
+                        get location() {
+                            return state.location;
                         }
                     });
                 });
